talk/basicconcepts/coresyntax.tex

```diff
@@ -47,51 +47,73 @@
   \begin{cppcode}
     bool b = true;            // boolean, true or false
 
-    char c = 'a';             // 8 bit ASCII char
+    char c = 'a';         // min 8 bit integer (signed or not)
+                              // can store an ASCII character
+    signed char c = 4;        // min 8 bit signed integer
+    unsigned char c = 4;      // min 8 bit unsigned integer
+
     char* s = "a C string";   // array of chars ended by \0
     string t = "a C++ string";// class provided by the STL
 
-    char c = -3;              // 8 bit integer (signed or not)
-    unsigned char c = 4;      // 8 bit unsigned integer
-
-    short int s = -444;       // 16 bit signed integer
-    unsigned short s = 444;   // 16 bit unsigned integer
+    short int s = -444;       // min 16 bit signed integer
+    unsigned short s = 444;   // min 16 bit unsigned integer
     short s = -444;           // int is optional
   \end{cppcode}
 \end{frame}
 \begin{frame}[fragile]
   \frametitlecpp[98]{Basic types(2)}
   \begin{cppcode}
-    int i = -123456;          // 32 bit signed integer
-    unsigned int i = 1234567; // 32 bit signed integer
+    int i = -123456;          // min 16, usually 32 bit
+    unsigned int i = 1234567; // min 16, usually 32 bit
 
-    long l = 0L               // 32 or 64 bits (ptr size)
-    unsigned long l = 0UL;    // 32 or 64 bits (ptr size)
+    long l = 0L               // min 32 bit
+    unsigned long l = 0UL;    // min 32 bit
```

Comment:
pointer size is only one factor, OS ABI is another one. Win64 has 64bit pointers, but 32bit `long`.